From f5e68928806dc5eb4d9fdc1de3829abe0d710aa3 Mon Sep 17 00:00:00 2001
From: Dezhen Wang <dezhen.wang@amlogic.com>
Date: Tue, 28 Oct 2025 08:50:14 +0000
Subject: [PATCH] upgrade: revert the bpf change from upstream [1/1]

PD#SWPL-231019

Problem:
upstream e7fb4ebee6e900899d2 change the initial value of jit_needed in
function bpf_prog_select_runtime. It lead to return a negative value when boot
up in amlogic arm kernel.

Solution:
revert the patch upstream e7fb4ebee6e900899d2

Verify:
local

Change-Id: Iac22c34bf5b61c1e3efcb486238dacf224ac8c22
Signed-off-by: Dezhen Wang <dezhen.wang@amlogic.com>
---
 kernel/bpf/core.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/kernel/bpf/core.c b/kernel/bpf/core.c
index 1ded3eb492b8e..b7255a7eba6e5 100644
--- a/kernel/bpf/core.c
+++ b/kernel/bpf/core.c
@@ -1909,7 +1909,12 @@ struct bpf_prog *bpf_prog_select_runtime(struct bpf_prog *fp, int *err)
 	/* In case of BPF to BPF calls, verifier did all the prep
 	 * work with regards to JITing, etc.
 	 */
+
+#ifdef CONFIG_AMLOGIC_ARMV8_AARCH32
+	bool jit_needed = false;
+#else
 	bool jit_needed = fp->jit_requested;
+#endif
 
 	if (fp->bpf_func)
 		goto finalize;
-- 
2.51.0

